cmd/xmtpd-cli/commands/node_registry.go

```diff
@@ -0,0 +1,521 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/xmtp/xmtpd/pkg/blockchain"
+	"github.com/xmtp/xmtpd/pkg/blockchain/migrator"
+	"github.com/xmtp/xmtpd/pkg/config"
+	"github.com/xmtp/xmtpd/pkg/utils"
+	"go.uber.org/zap"
+)
+
+func nodeRegistryCmd() *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "nodes",
+		Short: "Manage Node Registry",
+	}
+
+	cmd.PersistentFlags().
+		Uint32P("node-id", "n", 0, "Node ID to use")
+
+	cmd.AddCommand(
+		registerNodeCmd(),
+		canonicalNetworkCmd(),
+		getNodeCmd(),
+		maxCanonicalCmd(),
+		setHttpAddressCmd(),
+	)
+
+	return &cmd
+}
+
+func registerNodeCmd() *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "register",
+		Short: "Register a node",
+		Run:   registerNodeHandler,
+		Example: `
+Usage: xmtpd nodes register --owner-address <address> --signing-key-pub <key> --http-address <address> [--force]
```

Comment:
the cli command tends to be `xmtpd-cli` or maybe we need a different name. But historically it has not been `xmtpd`